app/src/pages/playground/playgroundUtils.ts

```diff
@@ -190,25 +190,13 @@ function processAttributeMessagesToChatMessage({
   provider: ModelProvider;
 }): ChatMessage[] {
   return messages.map(({ message }) => {
-    if (Array.isArray(message.content)) {
-      // check if message matches anthropic tool result schema
-      const { data, success } =
-        anthropicToolResultMessageSchema.safeParse(message);
-      const toolResult = data?.content.find(
-        (content) => content?.type === "tool_result"
-      );
-      if (success && toolResult != null) {
-        return {
-          id: generateMessageId(),
-          role: getChatRole("tool"),
-          content: toolResult.content,
-          toolCallId: toolResult.tool_use_id,
-        };
-      }
-    }
     return {
       id: generateMessageId(),
-      role: getChatRole(message.role),
+      // if the message has a tool call id, it is a tool "role" message from the perspective of the playground
+      role:
+        message.tool_call_id != null
+          ? getChatRole("tool")
```

Comment:
oh nice this works because we expect there to be just one tool result so the find previously on line 197 was always just getting the first content anyways which is a string?

Reply:
The find previously was not even being invoked because we do not have content instrumented as an array, nor a json stringified array.

It was a hold over from when we were attempting to support arbitrary json as content which is no longer a goal